packages/react/src/components/LocaleModal/LocaleModal.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { settings as ddsSettings, altlangs } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import LocaleModalRegions from './LocaleModalRegions';
+import LocaleModalCountries from './LocaleModalCountries';
+import { ArrowLeft20, Globe20 } from '@carbon/icons-react';
+import { ComposedModal, ModalHeader, ModalBody } from 'carbon-components-react';
+import { LocaleAPI } from '@carbon/ibmdotcom-services';
+import cx from 'classnames';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * LocaleModal component
+ *
+ * @param {object} props props object
+ * @param {boolean} props.isOpen Opens modal
+ * @param {boolean} props.setIsOpen isOpen state of modal
+ * @param {string} props.headerLabel modal header label
+ * @param {string} props.headerTitle modal header title
+ * @returns {*} LocaleModal component
+ */
+const LocaleModal = ({ isOpen, setIsOpen, ...localeModalProps }) => {
+  const [list, setList] = useState({});
+  const [isFiltering, setIsFiltering] = useState(false);
+  const [currentRegion, setCurrentRegion] = useState();
+
+  const filterClass = cx({
+    [`${prefix}--locale-modal__filtering`]: isFiltering,
+  });
+
+  useEffect(() => {
+    (async () => {
+      const locale = await LocaleAPI.getLocale();
+      const list = locale && (await LocaleAPI.getList(locale));
+      setList(list);
+    })();
+  }, []);
+
+  /**
+   *  New region/country list based lang attributes available on page
+   *
+   * @param {object} list country list
+   *
+   * @returns {object} list item
+   */
+  const sortList = list => {
+    const pageLangs = altlangs();
+    const filterList = [];
+
+    list.regionList &&
+      list.regionList.map((region, index) => {
+        let isAmericas = region.name.indexOf('America') > -1;
```

Comment:
@kennylam I think this would only work for the english translated countrylist, right? `America` is translated into the corresponding language in other countrylist files.

Reply:
@jeffchew At the moment, yes. Not sure how we're going to match 'North America` and 'Latin America' in dozens of languages to merge the lists. Been giving it some thought but haven't come up with a good solution yet.

Reply:
@jeffchew actually one thing I can do is change it to match region keys instead, `na` and `la`

Reply:
One thing I would even do is to not worry about regions, and just have the front-end be a dumb interface that iterates through the regions. We will eventually go back to the data and fix the data to have NA and LA to be merged together, I don't think that the UI layer should be doing that logic. I believe I already merged usen to have NA and LA merged already on the data side (unless you're pointing to an older data file), and we will round back to the other country list files and merge those later.

Reply:
@kennylam https://1.www.s81c.com/common/js/dynamicnav/www/countrylist/jsononly/usen-utf8.json